master/internal/trial.go

```diff
@@ -676,6 +676,13 @@ func (t *trial) handleAllocationExit(exit *task.AllocationExited) error {
 			}
 		}
 
+		if err := logpattern.ClearSignal(context.TODO(), t.taskID); err != nil {
+			return t.transition(model.StateWithReason{
+				State:               model.ErrorState,
+				InformationalReason: err.Error(),
+			})
+		}
+
```

Comment:
Can we please add some logs explaining why we want to clear signal when allocation exit fails

Reply:
I don't follow why we would want to clear it. It seems the point of this change is to track when some exceptional event happen and present them in a more user friendly manner. Losing that we had an ECC error on node xyz002 seems important to surface still.

Reply:
If a trial runs into an OOM error, it's caught in log and a `signal` will be shown in the UI. I think the `signal` only appears for a brief moment since the trial will restart quickly. This means user might not see there was an OOM error. You are right. I feel I shouldn't clear it at restart.